webhooks-extension/endpoints/webhook.go

```diff
@@ -126,12 +148,142 @@ func (r Resource) createWebhook(request *restful.Request, response *restful.Resp
 		return
 	}
 	webhooks[webhook.Name] = webhook
+	logging.Log.Debugf("Writing the GitHubSource webhook ConfigMap in namespace %s", installNs)
 	r.writeGitHubWebhooks(installNs, webhooks)
 	response.WriteHeader(http.StatusCreated)
 }
 
+// Removes from ConfigMap, removes the actual GitHubSource, removes the webhook
+func (r Resource) deleteWebhook(request *restful.Request, response *restful.Response) {
+	logging.Log.Debug("In deleteWebhook")
+	name := request.PathParameter("name")
+	namespace := request.QueryParameter("namespace")
+	deletePipelineRuns := request.QueryParameter("deletepipelineruns")
+
+	var toDeletePipelineRuns = false
+	var err error
+
+	if deletePipelineRuns != "" {
+		toDeletePipelineRuns, err = strconv.ParseBool(deletePipelineRuns)
+		if err != nil {
+			theError := errors.New("bad request information provided, cannot handle deletepipelineruns query (should be set to true or not provided)")
+			logging.Log.Error(theError)
+			RespondError(response, err, http.StatusInternalServerError)
+		}
+	}
+
+	if namespace == "" {
+		theError := errors.New("no namespace was provided")
+		logging.Log.Error(theError)
+		RespondError(response, theError, http.StatusBadRequest)
+	}
+
+	if name != "" {
+		foundRepoURL := r.findRepoURLFromConfigMap(name, namespace)
+		// This will remove any PipelineRuns too if specified to, hence the need for the repository URL
+		err = r.deleteGitHubWebhookByName(name, namespace, foundRepoURL, toDeletePipelineRuns)
```

Comment:
If I'm not mistaken, all of the Github webhooks are created in the install namespace (this is because they need to be in the same namespace as their sink). So, I think this `deleteGitHubWebhookByName` should exclusively delete from the install namespace.

Reply:
I would partly argue for leaving in the namespace arg as giving more future potential ..... it also allows for deleting when a sink is in a different namespace to the install namespace??  Depends on what install namespace is defined as .... presumably the install namespace of the dashboard

Reply:
I'll leave in for future potential as we want to work with various namespaces eventually